### PR TITLE
Add mechanism to process l2a level as well.

### PR DESCRIPTION
**Pull request by nekhtiari**
Add a mechanism to process level 2 imagery, e2e test and manifest are updated.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -16,3 +16,4 @@ pytest-mypy
 mock
 black
 coverage-badge
+google-auth>=1.19.*
```
Comment (rodrigoalmeida94) [inline requirements.txt:19]: Hmm do we need this in requirements? or is it for the e2e tests? I can see it causing some issues in CI

File: src/inference.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def check_size(dims):
             sys.exit(1)
 
     def start(self, path_to_input_img, path_to_output_img):
-        data_list = self.get_data(path_to_input_img)
+        data_list, image_level = self.get_data(path_to_input_img)
```
Comment (rodrigoalmeida94) [inline src/inference.py:55]: Awesome!

File: src/s2_tiles_supres.py [hunk 3]
```diff
@@ -102,10 +102,13 @@ def get_data(self, image_id) -> list:
         ):
             data_path = file
 
+        # The following line will define whether image is L1C or L2A
+        # For instance image_level can be "MSIL1C" or "MSIL2A"
+        image_level = Path(data_path).stem.split("_")[1]
```
Comment (rodrigoalmeida94) [inline src/s2_tiles_supres.py:107]: Perfect!

**Review by rodrigoalmeida94 — APPROVED** [review]
LGTM!! Awesome work at lightning speed ⚡

File: requirements.txt (hunk 1 above)
Comment (nekhtiari) [reply requirements.txt:19]: I guess we do not need it, I  was getting this error `ERROR: google-api-core 1.22.1 has requirement google-auth<2.0dev,>=1.19.1, but you'll have google-auth 1.14.3 which is incompatible.` but without it is also working.

File: src/supres.py [hunk 4]
```diff
@@ -49,18 +57,38 @@ def dsen2_60(d10, d20, d60):
...
+
+
 def _predict(test, model_filename):
-    # create model
     with STRATEGY.scope():
         model = keras.models.load_model(model_filename)
     LOGGER.info("Symbolic Model Created.")
     LOGGER.info("Predicting using file: %s", model_filename)
-    prediction = model.predict(test, verbose=1)
+    p = 0
```
Comment (nekhtiari) [inline src/supres.py:84]: what is p here?

File: tests/test_supres.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+"""
+This module include multiple test cases to check the performance of the s2_tiles_supres script.
+"""
+import tensorflow as tf
+import numpy as np
+import pytest
+from context import dsen2_60, dsen2_20, batch_generator, patches
+
+DISABLE_NO_GPU = pytest.mark.skipif(
```
Comment (nekhtiari) [inline tests/test_supres.py:9]: Nice!

File: src/supres.py (hunk 4 above)
Comment (rodrigoalmeida94) [reply src/supres.py:84]: just a counter, I use it so I don't have to initialize the prediction array in advance, in the first run I just predict, and then I append to the prediction

Comment (nekhtiari) [reply src/supres.py:84]: I see, thank you for the explanation.

**Comment by rodrigoalmeida94** [comment]
@nekhtiari with the 128 batch size we resolve the memory issue and we get a 15min run time per scene (from 20min with the previous block version). Testing now with multiple scenes.

File: e2e.py [hunk 6]
```diff
@@ -52,31 +52,37 @@ def assert_e2e():
...
 
 
-if __name__ == "__main__":
-    TESTNAME = "e2e_s2-superresolution"
-    TEST_DIR = Path("/tmp") / TESTNAME
-    TEST_DIR.mkdir(parents=True, exist_ok=True)
-    INPUT_DIR = TEST_DIR / "input"
-    FILES_TO_DELETE = Path(TEST_DIR / "output").glob("*")
-    for file_path in FILES_TO_DELETE:
+def setup(testname):
```
Comment (markusuwe) [inline e2e.py:55]: Nice to cut the function into a setup and assertion part.

File: e2e.py [hunk 7]
```diff
@@ -11,17 +11,17 @@
 # WARNING
 # THIS E2E TEST WILL ONLY WORK IN GPU ENABLED MACHINES
 # pylint: disable=invalid-name
```
Comment (markusuwe) [inline e2e.py:13]: Any idea what causes this problem?

File: e2e_l2a.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+End-to-end test: Fetches data, creates output, stores it in /tmp and checks if output
+is valid.
+"""
+import os
+
+# WARNING
+# THIS E2E TEST WILL ONLY WORK IN GPU ENABLED MACHINES
+# pylint: disable=invalid-name
+
+from e2e import assert_e2e, setup
```
Comment (markusuwe) [inline e2e_l2a.py:11]: Great!

File: tests/test_patches.py [hunk 9]
```diff
@@ -30,15 +31,15 @@ def scale_60():
     return 6
 
 
-@pytest.mark.skip()
+@DISABLE_NO_GPU
```
Comment (markusuwe) [inline tests/test_patches.py:34]: That's cool!

**Review by markusuwe — COMMENTED** [review]
A few comments (apologies, the review was unasked for).
Want to spend a couple of hours today on S2 superres so I might be of more use if the problems don't go away soon.

**Comment by nekhtiari** [comment]
> @nekhtiari with the 128 batch size we resolve the memory issue and we get a 15min run time per scene (from 20min with the previous block version). Testing now with multiple scenes.

wow, So cool!

File: e2e.py (hunk 7 above)
Comment (rodrigoalmeida94) [reply e2e.py]: Removed! Just capital letter variables

File: src/supres.py (hunk 4 above)
Comment (rodrigoalmeida94) [reply src/supres.py]: Added as flag.

**Comment by rodrigoalmeida94** [comment]
@markusuwe @nekhtiari tested and working on all possible cases, small chunk takes only 2min to superresolve! Also applied your suggestions. I guess good to merge?

**Comment by nekhtiari** [comment]
> @markusuwe @nekhtiari tested and working on all possible cases, small chunk takes only 2min to superresolve! Also applied your suggestions. I guess good to merge?

Nice, yes, I guess we can!

**Pull request merged**